packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -248,10 +307,79 @@ - (void)errorNoPhotoAccess:(PHAuthorizationStatus)status {
 
 - (void)showPhotoLibrary {
   // No need to check if SourceType is available. It always is.
-  _imagePickerController.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
-  [[self viewControllerWithWindow:nil] presentViewController:_imagePickerController
-                                                    animated:YES
-                                                  completion:nil];
+  if (_phPickerFlag) {
+    [[self viewControllerWithWindow:nil] presentViewController:_pickerViewController
+                                                      animated:YES
+                                                    completion:nil];
+  } else {
+    _imagePickerController.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+    [[self viewControllerWithWindow:nil] presentViewController:_imagePickerController
+                                                      animated:YES
+                                                    completion:nil];
+  }
+}
+
+// Limited access to the photo library
+- (void)showLimitedPhotoLibrary {
+  [[PHPhotoLibrary sharedPhotoLibrary]
+      presentLimitedLibraryPickerFromViewController:_pickerViewController];
+}
+
+- (void)picker:(PHPickerViewController *)picker
+    didFinishPicking:(NSArray<PHPickerResult *> *)results API_AVAILABLE(ios(14)) {
+  [picker dismissViewControllerAnimated:YES completion:nil];
+
+  for (PHPickerResult *result in results) {
+    [result.itemProvider
+        loadObjectOfClass:[UIImage class]
+        completionHandler:^(__kindof id<NSItemProviderReading> _Nullable image,
+                            NSError *_Nullable error) {
+          if ([image isKindOfClass:[UIImage class]]) {
+            if (image != nil) {
+              NSNumber *maxWidth = [self->_arguments objectForKey:@"maxWidth"];
```

Comment:
While you are here, could you refactor _arguments into a property then use weak self. :)